### PR TITLE
Use intrinsic for GetHighestSetBitIndex

### PR DESCRIPTION
**Pull request by benaadams**
## Changes

- Use `BitOperations.LeadingZeroCount` rather than a bunch of `if`s

#### What types of changes does your code introduce?

- [x] Optimization

#### Requires testing

- [x] No

### CODE REVIEW

**Comment by smartprogrammer93** [comment]
Benchmarks?

**Review by LukaszRozmej — APPROVED** [review]

File: src/Nethermind/Nethermind.Core/Extensions/Bytes.cs [hunk 1]
```diff
@@ -145,20 +145,12 @@ public static bool GetBit(this byte b, int bitNumber)
...
-        {
-            if ((b & 128) == 128) return 8;
-            if ((b & 64) == 64) return 7;
-            if ((b & 32) == 32) return 6;
-            if ((b & 16) == 16) return 5;
-            if ((b & 8) == 8) return 4;
-            if ((b & 4) == 4) return 3;
-            return (b & 2) == 2 ? 2 : b;
-        }
+            => 32 - BitOperations.LeadingZeroCount(b);
```
[the remark this answers is not in the document]
Comment (benaadams) [reply src/Nethermind/Nethermind.Core/Extensions/Bytes.cs:153]: LeadingZeroCount takes a `uint` so upcasts; hence 32. Is well tested with `BytesTests.Can_get_highest_bit_set`

**Comment by benaadams** [comment]
> Benchmarks?

Is single instruction `lzcnt` vs 6 conditional branches

Before
```asm
G_M000_IG03:                ;; offset=0x0010
       movzx    rcx, bl
       test     cl, 128
       je       SHORT G_M000_IG06

G_M000_IG04:                ;; offset=0x0018
       mov      eax, 8

G_M000_IG05:                ;; offset=0x001D
       add      rsp, 32
       pop      rbx
       ret      

G_M000_IG06:                ;; offset=0x0023
       test     cl, 64
       je       SHORT G_M000_IG08
       mov      eax, 7

G_M000_IG07:                ;; offset=0x002D
       add      rsp, 32
       pop      rbx
       ret      

G_M000_IG08:                ;; offset=0x0033
       test     cl, 32
       je       SHORT G_M000_IG10
       mov      eax, 6

G_M000_IG09:                ;; offset=0x003D
       add      rsp, 32
       pop      rbx
       ret      

G_M000_IG10:                ;; offset=0x0043
       test     cl, 16
       je       SHORT G_M000_IG11
       mov      eax, 5
       jmp      SHORT G_M000_IG14

G_M000_IG11:                ;; offset=0x004F
       test     cl, 8
       je       SHORT G_M000_IG12
       mov      eax, 4
       jmp      SHORT G_M000_IG14

G_M000_IG12:                ;; offset=0x005B
       test     cl, 4
       je       SHORT G_M000_IG13
       mov      eax, 3
       jmp      SHORT G_M000_IG14

G_M000_IG13:                ;; offset=0x0067
       mov      eax, 2
       test     cl, 2
       cmove    eax, ecx

G_M000_IG14:                ;; offset=0x0072
       add      rsp, 32
       pop      rbx
       ret
```
After
```asm
G_M000_IG03:                ;; offset=0x0010
       movzx    rax, bl
       lzcnt    eax, eax
       neg      eax
       add      eax, 32

G_M000_IG04:                ;; offset=0x001C
       add      rsp, 32
       pop      rbx
       ret      
```

**Review by smartprogrammer93 — APPROVED** [review]

**Review by LukaszRozmej — APPROVED** [review]

File: src/Nethermind/Nethermind.Core/Extensions/Bytes.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (benaadams) [reply src/Nethermind/Nethermind.Core/Extensions/Bytes.cs]: Added explicit cast so more obvious

**Pull request merged**